aes/src/x86/vaes512/aes256.rs

```diff
@@ -0,0 +1,687 @@
+use crate::x86::{arch::*, RoundKeys, Simd512RoundKeys};
+use crate::Block64;
+use cipher::inout::InOut;
+use core::{arch::asm, mem::MaybeUninit};
+
+#[inline(always)]
+pub(crate) unsafe fn parallelize_keys(keys: &RoundKeys<15>) -> Simd512RoundKeys<15> {
+    let mut v512: [MaybeUninit<__m512i>; 15] = MaybeUninit::uninit().assume_init();
+    asm! {
+        "vbroadcasti32x4 zmm0 , [{keys} +  0 * 16]",
+        "vbroadcasti32x4 zmm1 , [{keys} +  1 * 16]",
+        "vbroadcasti32x4 zmm2 , [{keys} +  2 * 16]",
+        "vbroadcasti32x4 zmm3 , [{keys} +  3 * 16]",
+        "vbroadcasti32x4 zmm4 , [{keys} +  4 * 16]",
+        "vbroadcasti32x4 zmm5 , [{keys} +  5 * 16]",
+        "vbroadcasti32x4 zmm6 , [{keys} +  6 * 16]",
+        "vbroadcasti32x4 zmm7 , [{keys} +  7 * 16]",
+        "vbroadcasti32x4 zmm8 , [{keys} +  8 * 16]",
+        "vbroadcasti32x4 zmm9 , [{keys} +  9 * 16]",
+        "vbroadcasti32x4 zmm10, [{keys} + 10 * 16]",
+        "vbroadcasti32x4 zmm11, [{keys} + 11 * 16]",
+        "vbroadcasti32x4 zmm12, [{keys} + 12 * 16]",
+        "vbroadcasti32x4 zmm13, [{keys} + 13 * 16]",
+        "vbroadcasti32x4 zmm14, [{keys} + 14 * 16]",
+
+        "vmovdqu32 [{optr} +  0 * 64], zmm0",
+        "vmovdqu32 [{optr} +  1 * 64], zmm1",
+        "vmovdqu32 [{optr} +  2 * 64], zmm2",
+        "vmovdqu32 [{optr} +  3 * 64], zmm3",
+        "vmovdqu32 [{optr} +  4 * 64], zmm4",
+        "vmovdqu32 [{optr} +  5 * 64], zmm5",
+        "vmovdqu32 [{optr} +  6 * 64], zmm6",
+        "vmovdqu32 [{optr} +  7 * 64], zmm7",
+        "vmovdqu32 [{optr} +  8 * 64], zmm8",
+        "vmovdqu32 [{optr} +  9 * 64], zmm9",
+        "vmovdqu32 [{optr} + 10 * 64], zmm10",
+        "vmovdqu32 [{optr} + 11 * 64], zmm11",
+        "vmovdqu32 [{optr} + 12 * 64], zmm12",
+        "vmovdqu32 [{optr} + 13 * 64], zmm13",
+        "vmovdqu32 [{optr} + 14 * 64], zmm14",
+
+        keys = in(reg) keys.as_ptr(),
+        optr = in(reg) v512.as_mut_ptr().cast::<__m512i>(),
+
+        clobber_abi("C"),
+        options(nostack, preserves_flags),
+    };
+    core::mem::transmute(v512)
+}
+
+#[inline]
+pub(crate) unsafe fn encrypt64(keys: &Simd512RoundKeys<15>, blocks: InOut<'_, '_, Block64>) {
+    let (iptr, optr) = blocks.into_raw();
+    asm! {
+        // load keys
+        "vmovdqu32 zmm0 , [{keys} +  0 * 64]",
```

Comment:
Since we reload the keys on each call either way, it may be worth to increase number of blocks processed in parallel to 124 and interleave loading like you did for vaes256.